core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteractionEditorDirective.js

```diff
@@ -97,19 +105,16 @@ oppia.directive('stateInteractionEditor', [
 
           $scope.$on('stateEditorInitialized', function(evt, stateData) {
             $scope.hasLoaded = false;
-
             InteractionDetailsCacheService.reset();
 
-            $scope.stateName = EditorStateService.getActiveStateName();
-
+            $scope.stateName = stateData.name;
```

Comment:
Why have both this and getStateName() (above)?

Reply:
Also there's stateContentService.stateName below, so in effect you're getting stateName from three different sources. It would be better to get it from just one if possible, and ideally this directive wouldn't even know what the state name is at all. Could the parent directive maintain an activeStateName variable, perhaps?

Reply:
The $scope.getStateName() returned undefined here when I tried, maybe because when the 'stateEditorInitialized' was broadcasted, the state name wouldn't have been initialized yet. 
I'll follow the comments below and remove stateName from these function calls.

Reply:
I still wonder if you can manage to get to the point where this directive doesn't need to know anything about stateName at all. Is that possible?